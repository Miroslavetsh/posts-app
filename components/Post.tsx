import Link from 'next/link'
import styled from 'styled-components'

import IPost from '../models/Post'

export interface PostPropTypes {
  post: IPost
}

interface StyledPostPropTypes {
  textColor: string
}

const StyledPost = styled.div<StyledPostPropTypes>`
  background: #000;
  color: ${({ textColor }) => (textColor ? textColor : '#000')}; ;
`

const Post: React.FC<PostPropTypes> = ({ post }) => {
  return (
    <StyledPost textColor='#fff'>
      <Link href={`/${post.id}`}>{post.title}</Link>
    </StyledPost>
  )
}

export default Post
