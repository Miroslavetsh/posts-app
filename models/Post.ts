export default interface IPost {
  userId: number
  id: number
  title: string
  body: string
}
