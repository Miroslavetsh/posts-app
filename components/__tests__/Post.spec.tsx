import React from 'react'
import { render } from 'enzyme'

import Post, { PostPropTypes } from '../Post'

const setUp = (props: PostPropTypes) => render(<Post {...props} />)

describe('Post component', () => {
  const props = {
    post: {
      userId: 1,
      id: 1,
      title: 'Uchiha Itachi',
      body: 'Body of Itachi',
    },
  }

  it('renders a div that contains an <a> tag with href and title in', () => {
    const wrapper = setUp(props)
    expect(wrapper).toMatchSnapshot()
  })

  it('contains the title of post from props', () => {
    const wrapper = setUp(props)
    expect(wrapper.find('a').text()).toEqual(props.post.title)
  })

  it('contains a link to the post with id equals id from props', () => {
    const wrapper = setUp(props)
    expect(wrapper.find('a').attr('href')).toContain(props.post.id.toString())
  })
})
