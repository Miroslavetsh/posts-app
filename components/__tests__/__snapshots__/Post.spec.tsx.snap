// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders a div that contains an <a> tag with href and title in 1`] = `
<div
  class="sc-bdvvtL eIoWgC"
>
  <a
    href="/1"
  >
    Uchiha Itachi
  </a>
</div>
`;
